django/core/mail/message.py

```diff
@@ -280,6 +282,8 @@ def message(self):
             # Use cached DNS_NAME for performance
             msg["Message-ID"] = make_msgid(domain=DNS_NAME)
         for name, value in self.extra_headers.items():
+            if name.lower() in HEADER_WITH_ONE_OCCURRENCE and msg[name] is not None:
+                raise ValueError("Cannot insert repeated headers")
             if name.lower() != "from":  # From is already handled
                 msg[name] = value
```

Comment:
An alternative to raising an exception, could be assigning the same way we do for `from`.

The behaviour change would then be that the headers in `extra_headers` will silently be ignored - it's probably safer to raise for users, so that they are made aware of the change.